wayland-server/src/event_sources.rs

```diff
@@ -50,6 +50,18 @@ impl<ID> FdEventSource<ID> {
         }
     }
 
+    /// Do something with the implementation data without removing it from the `EventLoop`
+    pub fn with_idata<F, R>(&mut self, evlh: &mut EventLoopHandle, f: F) -> R
+    where
+        R: 'static,
```

Comment:
Why is this `R: 'static` needed? (applies to all 4 functions)

Reply:
I did not want the user to return something bound to the lifetime of idata, which would likely be a reference to idata. But of course you could move something with a lifetime into the closure and try to return it.

Reply:
I *think* the Rust rules won't allow the lifetime of `R` to be bound to the lifetimes of either argument....

But even if it did, it would mutably borrow `self` and the event loop handle, preventing the user to do anything bad, wouldn't it?

Reply:
Yeah sounds like it. I am probably just paranoid.